packages/api/app/Middleware/HandleParam.js

```diff
@@ -0,0 +1,51 @@
+/** @typedef {import('@adonisjs/framework/src/Request')} Request */
+/** @typedef {import('@adonisjs/framework/src/Response')} Response */
+
+const Database = use('Database');
+
+class HandleParam {
+	/** JSDoc Block
+	 *
+	 * @param {object} ctx The content of the request
+	 * @param {Request} ctx.request The HTTP request
+	 * @param {Function} next next
+	 */
+	async handle({ request, response }, next) {
+		const data = request.only(['page', 'perPage', 'order', 'orderBy']);
+
+		data.order = data.order ? data.order.toLowerCase() : '';
+		data.orderBy = data.orderBy ? data.orderBy.toLowerCase() : '';
+		data.page = data.page > 0 ? data.page : 1;
+		data.perPage = data.perPage < 100 ? data.perPage : 10;
```

Comment:
Acho que essa condição está errada pois temos duas situaçãoes;

data.perPage não exsitir ele deve ser 10.
data.perPage > 100 deve ser limitado a 100.

Do jeito que está se eu passar perPage > 100 ele vai setar perPage para 10.

```javascript
const maxPerPage = 100;
data.perPage = data.perPage || 10;
data.perPage = data.perPage <= maxPerPage ? data.perPage : maxPerPage ;
```

Reply:
Eh verdade

Reply:
Nesse caso ai quando existir mas não for um numero, ele vai seta o perPage para 100

Reply:
Por isso tinha um `> 0`

Reply:
você pode fazer a conversão `Number(data.perPage)`